lib/message_bus/rack/middleware.rb

```diff
@@ -81,79 +81,93 @@ def handle_request(env)
     client_id = env['PATH_INFO'][@base_route_length..-1].split("/")[0]
     return [404, {}, ["not found"]] unless client_id
 
-    user_id = @bus.user_id_lookup.call(env) if @bus.user_id_lookup
-    group_ids = @bus.group_ids_lookup.call(env) if @bus.group_ids_lookup
-    site_id = @bus.site_id_lookup.call(env) if @bus.site_id_lookup
+    MessageBus.trace("middleware/authentication") do
+      user_id = @bus.user_id_lookup.call(env) if @bus.user_id_lookup
+      group_ids = @bus.group_ids_lookup.call(env) if @bus.group_ids_lookup
+      site_id = @bus.site_id_lookup.call(env) if @bus.site_id_lookup
+    end
 
     # close db connection as early as possible
     close_db_connection!
 
-    client = MessageBus::Client.new(message_bus: @bus, client_id: client_id,
-                                    user_id: user_id, site_id: site_id, group_ids: group_ids)
+    MessageBus.trace("middleware/subscriptions") do
```

Comment:
Should the metric name also include `messagebus` so that consumers don't need to add it?

```suggestion
    MessageBus.trace("message_bus/middleware/subscriptions") do
```